File: app/src/main/java/com/greglaun/lector/TextProvider.java
package com.greglaun.lector;

import java.util.List;

public interface TextProvider {
    String END_OF_STREAM = "com.greglaun.eof";
    String WIKI_BASE = "https://en.wikipedia.org/wiki/";


    String provideOneText();
    List<String> provideText(int n);

    boolean fastForwardTo(String place);

    List<String> provideAllTexts();

    String getTitle();

    String getHtml();
}

File: app/src/main/java/com/greglaun/lector/TextProvider.kt
package com.greglaun.lector

interface TextProvider {

    fun provideOneText(): String
    fun provideText(n: Int): List<String>

    fun fastForwardTo(place: String): Boolean

    fun provideAllTexts(): List<String>

    companion object {
        val END_OF_STREAM = "com.greglaun.eof"
        val WIKI_BASE = "https://en.wikipedia.org/wiki/"
        val title: String? = null
        val html: String? = null
    }
}
